dm/dm-precheck.md

```diff
@@ -1,92 +1,130 @@
+
+* (Required) Database primary-secondary configuration
+
+    - The database ID `server_id` must be configured for the upstream database (GTID is recommended for non-AWS Aurora environments).
+
+* (Required) MySQL binlog configuration
+
+    - Check whether binlog is enabled (DM requires binlog to be enabled).
+    - Check whether `binlog_format=ROW` is configured (DM only supports the migration of binlog in the ROW format).
+    - Check whether `binlog_row_image=FULL` is configured (DM only supports `binlog_row_image=FULL`).
+    - If `binlog_do_db` or `binlog_ignore_db` is configured, check whether the database tables to be migrated meet the conditions of `binlog_do_db` and `binlog_ignore_db`.
+
+* (Required) Check if the upstream database is in an [Online-DDL](/dm/feature-online-ddl.md) process (in which the `ghost` table is created but the `rename` phase is not executed yet). If the upstream is in the online-DDL process, the precheck returns errors. In this case, wait until the DDL to complete and retry.
+
+### Full and incremental data migration checking items
 
-    + The conflict of the auto increment primary keys in the sharded tables
+For the full and incremental data migration mode (`task-mode: all`), in addition to the [common checking items](#common-checking-items), the precheck also includes the [full data migration checking items](#full-data-migration-checking-items) and the [incremental data migration checking items](#incremental-data-migration-checking-items).
 
-        - The check fails in the following two conditions:
+## Configure precheck arguments
 
-            - The auto increment primary key exists in the sharded tables and its column type *is not* bigint.
-            - The auto increment primary key exists in the sharded tables and its column type *is* bigint, but column mapping *is not* configured.
+The migration task precheck supports processing in parallel. Even if the number of rows in sharded tables reaches a million level, the check can be completed in minutes.
 
-        - The check succeeds in other conditions except the two above.
+To specify the number of threads for the precheck, you can configure the `threads` argument of the `mydumpers` field in the migration task configuration file.
 
-### Disable checking items
+```yaml
+mydumpers:                           # Configuration arguments of the dump processing unit
+  global:                            # Configuration name
+    threads: 4                       # The number of threads that access the upstream when the dump processing unit performs the precheck and exports data from the upstream database (4 by default)
+    chunk-filesize: 64               # The size of the file generated by the dump processing unit (64 in MB by default)
+    extra-args: "--consistency none" # Other arguments of the dump processing unit. You do not need to manually configure table-list in `extra-args`, because table-list is automatically generated by DM.
```

Comment:
```suggestion
    extra-args: "--consistency none" # Other arguments of the dump processing unit. You do not need to manually configure table-list in `extra-args`, because it is automatically generated by DM.
```